cmd/util/ledger/reporters/export_reporter.go

```diff
@@ -0,0 +1,61 @@
+package reporters
+
+import (
+	"encoding/json"
+	"io/ioutil"
+
+	"github.com/rs/zerolog"
+
+	"github.com/onflow/flow-go/ledger"
+	"github.com/onflow/flow-go/model/flow"
+)
+
+type ExportReport struct {
+	CurrentStateCommitment  string
+	EpochCounter            uint64
+	PreviousStateCommitment flow.StateCommitment
+}
+
+// ExportReporter writes data that can be leveraged outside of extraction
+type ExportReporter struct {
+	Chain                   flow.Chain
+	Log                     zerolog.Logger
+	PreviousStateCommitment flow.StateCommitment
+}
+
+func (e *ExportReporter) Name() string {
+	return "ExportReporter"
+}
+
+func (e *ExportReporter) Report(payload []ledger.Payload, o ledger.ExportOutputs) error {
+	script, _, err := ExecuteCurrentEpochScript(e.Chain, payload)
+
+	if err == nil && script.Err == nil && script.Value != nil {
```

Comment:
I recommend inverting this if statement and exiting early in case we failed to get the epoch counter. 

That way the else statement isn't needed. So in short:

```
if not_good {
// log and return
}
// report the good stuff
```

Reply:
I agree. This is poorly structured. I will update this to remove the need for the else clause.